src/sys/socket/mod.rs

```diff
@@ -61,6 +61,16 @@ pub enum SockType {
     Rdm = libc::SOCK_RDM,
 }
 
+#[derive(Clone, Copy, PartialEq, Eq, Debug)]
+#[repr(i32)]
+pub enum SockProtocol {
+    Default = 0,
```

Comment:
Seems like this shouldn't exist and really `socket()` should take an `Option<SockProtocol>`.

Also this enum and all it's variants needs docs. Be sure to link to relevant manpages!